File: Inversion/parallel/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/flib/pltlgd.f
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: PLTLGD                                                c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
        subroutine pltlgd(x,y,n,x1,y1,deltax,deltay,lintyp,inteq,
     1      ht,nocx,nocy,ipat,xlen)
        real*4 x(1),y(1)
c-----
c       plot y(x) in linear, log or semilog plots. Check
c       for bounds of plot
c       ipat    integer providing bit pattern for dashed lines
c       xlen    length of each on bit in pattern ipat
c-----
        if(ht.le.0.0)then
            hht = 0.07
        else
            hht = ht
        endif
        if(nocx.gt.0)then
            xmin = 10.**x1
            xmax = 10.**(x1+nocx)
        endif
        if(nocy.gt.0)then
            ymin = 10.**y1
            ymax = 10.**(y1+nocy)
        endif
c-----
c       plot, checking that bounds are not exceeded in log plots
c-----
        ipen = 3
        do 100 i=1,n
            if(nocx.gt.0)then
                if(x(i).ge.xmin .and. x(i).le.xmax)then
                    xx = (alog10(x(i))-x1)*deltax
                elseif(x(i).lt.xmin)then
                    xx = 0.0
                else
                    xx =  nocx * deltax
                endif
            else
                xx = (x(i)-x1)/deltax
            endif
            if(nocy.gt.0)then
                if(y(i).ge.ymin .and. y(i).le.ymax)then
                    yy = (alog10(y(i))-y1)*deltay
                else if(y(i).lt.ymin)then
                    yy = 0.0
                else
                    yy =   deltay *nocy
                endif
            else
                yy = (y(i)-y1)/deltay
            endif
            if(ipat.eq.0 .or. ipen.eq.3)then
                call plot(xx,yy,ipen)
            else
                call plotd(xx,yy,ipat,xlen)
            endif
            if(lintyp.ne.0)call symbol(xx,yy,hht,char(inteq),0.0,-1)
            if(lintyp.ge.0)ipen=2
  100   continue
        return
        end

File: Inversion/parallel/NA/src/hv_subs/PROGRAMS.330/CALPLOT/src/clib/pltlgd.c
/*
c---------------------------------------------------------------------c
c                                                                     c
c      COMPUTER PROGRAMS IN SEISMOLOGY                                c
c      VOLUME I                                                       c
c                                                                     c
c      PROGRAM: PLTLGD                                                c
c                                                                     c
c      COPYRIGHT (C)  1997 R. B. Herrmann                             c
c                                                                     c
c      Department of Earth and Atmospheric Sciences                   c
c      Saint Louis University                                         c
c      3507 Laclede Avenue                                            c
c      St. Louis, Missouri 63103                                      c
c      U. S. A.                                                       c
c                                                                     c
c---------------------------------------------------------------------c
*/

#include <math.h>
#include "calplot.h"

void pltlgd(float *x,float *y,int n,float x1,float y1,
		float deltax,float deltay,int lintyp,int inteq,
		float ht,int nocx,int nocy,int ipat,float xlen)
{
/*
c-----
c	plot y(x) in linear, log or semilog plots. Check
c	for bounds of plot
c	ipat	integer providing bit pattern for dashed lines
c	xlen	length of each on bit in pattern ipat
c-----
*/
	float hht, xmin, xmax, xx, yy;
	float ymin, ymax;
	int i, ipen;
	char c[4];

	if(ht <= 0.0){
		hht = 0.07;
	} else {
		hht = ht;
	}
	if(nocx > 0){
		xmin = pow(10.0,x1);
		xmax = pow(10.0,(x1+(float)nocx));
	}
	if(nocy > 0){
		ymin = pow(10.0,y1);
		ymax = pow(10.0,(y1+(float)nocy));
	}
/*
c-----
c	plot, checking that bounds are not exceeded in log plots
c-----
*/
	ipen = 3;
	c[0] = (char)inteq;
	for(i=0;i<n;i++){
		if(nocx > 0){
			if(x[i] >= xmin  &&  x[i] <= xmax){
				xx = (log10(x[i])-x1)*deltax;
			} else if(x[i] < xmin){
				xx = 0.0;
			} else {
				xx =  nocx * deltax;
			}
		} else {
			xx = (x[i]-x1)/deltax;
		}
		if(nocy > 0){
			if(y[i] >= ymin  &&  y[i] <= ymax){
				yy = (log10(y[i])-y1)*deltay;
			} else if(y[i] < ymin){
				yy = 0.0;
			} else {
				yy =   deltay *nocy;
			}
		} else {
			yy = (y[i]-y1)/deltay;
		}
		if(ipat == 0 || ipen == 3){
			plot(xx,yy,ipen);
		} else {
			plotd(xx,yy,ipat,xlen);
		}
		if(lintyp != 0)symbol(xx,yy,hht,c,0.0,-1);
		if(lintyp >= 0)ipen=2;
	}
	return;
}
